src/components/campaigns/CampaignMessages.tsx

```diff
@@ -33,40 +35,62 @@ const StyledGrid = styled(Grid)(({ theme }) => ({
     alignItems: 'flex-start',
     marginBottom: theme.spacing(2),
   },
+  [`& .${classes.pagination}`]: {
+    margin: theme.spacing(4, 0),
+  },
 }))
 
+const pageSize = 12
+
 export default function CampaignMessages({ campaignId }: { campaignId: string }) {
+  const [pageIndex, setPageIndex] = useState<number>(0)
   const { t, i18n } = useTranslation()
   const locale = i18n.language == 'bg' ? bg : enUS
-  const { data: list } = useDonationWishesList(campaignId)
+  const { data, isSuccess } = useDonationWishesList(campaignId, pageIndex, pageSize)
+  const numOfPages = isSuccess ? Math.ceil(data.totalCount / pageSize) : undefined
+
+  function handlePageChange(_e: React.ChangeEvent<unknown>, page: number) {
+    // <Pagination /> 's impl is 1 index based
+    // Our pagination apis are 0 index based
+    setPageIndex(page - 1)
+  }
 
   return (
     <StyledGrid className={classes.messagesWrapper}>
       <Grid className={classes.messagesTitleWrapper}>
         <RateReviewIcon color="action" />
         <Typography variant="h6">{t('campaigns:campaign.messages')}</Typography>
       </Grid>
-      {list?.map((wish) => (
-        <Grid item container key={wish.id} className={classes.wishWrapper}>
-          <Grid item xs={12}>
-            <AccountCircleIcon
-              sx={{ fontSize: '1.8rem', mr: '6px', mb: '-11px' }}
-              color="disabled"
-            />
-            <span>
-              {wish.person
-                ? wish.person.firstName + ' ' + wish.person.lastName
-                : t('campaigns:donations.anonymous')}
-            </span>
-          </Grid>
-          <Grid item xs={12} pl={'33px'}>
-            <Typography>
-              <q>{wish.message}</q>
-            </Typography>
-            <Typography>{getExactDate(wish.createdAt, locale)}</Typography>
+      {isSuccess && numOfPages! > 1 && (
```

Comment:
The intention here would be more explicit if we make it as:

```suggestion
      {isSuccess && numOfPages && numOfPages > 1 && (
```